esp-hal/src/soc/esp32/radio_clocks.rs

```diff
@@ -11,13 +11,13 @@
 //! `RadioClockControl` struct. This trait provides methods to enable, disable,
 //! reset the MAC, initialize clocks and perform other related operations.
 
-use crate::system::{RadioClockControl, RadioClockController, RadioPeripherals};
+use crate::system::{RadioClockController, RadioPeripherals};
 
 const DPORT_WIFI_CLK_WIFI_BT_COMMON_M: u32 = 0x000003c9;
 const DPORT_WIFI_CLK_WIFI_EN_M: u32 = 0x00000406;
 const DPORT_WIFI_CLK_BT_EN_M: u32 = 0x00030800;
 
-impl RadioClockController for RadioClockControl {
+impl RadioClockController for crate::peripherals::RADIO_CLK {
```

Comment:
Could we import this (for rest of the chips as well) instead of using whole path?

Reply:
It's only referenced once in the file, what do we gain by importing it a few lines above rather than just referencing its full path here?

Reply:
I rushed with writing my comment as it's more of my personal preference, haha, sorry, I guess you are right here.

Reply:
it's probably a matter of taste but I personally try to avoid importing types which are used only once (or maybe twice)

I had a brief look at Clippy lints but seems nothing there would meet that requirement (the closest is  https://rust-lang.github.io/rust-clippy/master/index.html#/absolute_paths but not really what we want)